runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -195,23 +193,24 @@ public void computeOutputs() {
     throw new IllegalStateException("Cannot resolve un-known PObject: " + value);
   }
 
-  @Override
-  public <T> T getAggregatorValue(String named, Class<T> resultType) {
-    return runtime.getAggregatorValue(AccumulatorSingleton.getInstance(jsc), named, resultType);
-  }
-
-  @Override
   public <T> AggregatorValues<T> getAggregatorValues(Aggregator<?, T> aggregator)
       throws AggregatorRetrievalException {
     return runtime.getAggregatorValues(AccumulatorSingleton.getInstance(jsc), aggregator);
   }
 
-  @Override
-  public MetricResults metrics() {
-    throw new UnsupportedOperationException("The SparkRunner does not currently support metrics.");
+  public <T> T getAggregatorValue(String named, Class<T> resultType) {
```

Comment:
I understand what you're saying, but I think this may introduce some less than ideal coupling between `SparkPipelineResult` and the runtime context, i.e., `SparkRuntimeContext`, in addition to the existing coupling to `EvaluationContext`.

The current design is meant to hide `SparkRuntimeContext` from `SparkPipelineResult` by exposing the needed services via `EvaluationContext`, i.e., the two overloads of the `getAggregatorValue` method. 

I was trying to avoid having stuff like `context.getRuntimeContext().getAggregatorValues(..)` in `SparkPipelineResult` since it would make `SparkPipelineResult` know about the runtime context (which I referred to as feature envy).